app/src/main/java/com/xinto/mauth/MainActivity.kt

```diff
@@ -49,22 +55,79 @@ class MainActivity : ComponentActivity() {
             }
             .launchIn(lifecycleScope)
 
+        // Parse Account to add, if there is one to add
+        var accountToAdd: DomainAccountInfo? = null
+        var parseError: OtpUriParserError? = null
+
+
+        if (intent?.scheme == "otpauth" && intent?.dataString != null) {
+            val parser = OtpUriParserImpl()
+
+            accountToAdd = when (val result = parser.parseOtpUri(intent?.dataString!!)) {
+                is OtpUriParserResult.Success ->
+                    DomainAccountInfo(
+                        id = null,
+                        icon = null,
+                        label = result.data.label,
+                        issuer = result.data.issuer,
+                        secret = result.data.secret,
+                        algorithm = result.data.algorithm,
+                        type = result.data.type,
+                        digits = result.data.digits,
+                        counter = result.data.counter ?: 0,
+                        period = result.data.period ?: 30,
+                    )
+                is OtpUriParserResult.Failure -> {
+                    parseError = result.error
+                    null
+                }
+            }
+        }
+
         setContent {
             MauthTheme {
                 Surface(
                     modifier = Modifier.fillMaxSize(),
                     color = MaterialTheme.colorScheme.background
                 ) {
-                    Main()
+                    // If there was an error parsing the URI, display an alert dialog
+                    if (parseError != null) {
+                        val showErrorDialog = remember { mutableStateOf(true) }
```

Comment:
This could be changed to use the State delegate
```suggestion
                        var showErrorDialog by remember { mutableStateOf(true) }
```

Reply:
Update the `.value` calls respectively